SU2_CFD/src/solvers/CSolver.cpp

```diff
@@ -4265,74 +4268,148 @@ void CSolver::LoadInletProfile(CGeometry **geometry,
           /*--- Get data for this profile. ---*/
 
           vector<passivedouble> Inlet_Data = profileReader.GetDataForProfile(jMarker);
-
           unsigned short nColumns = profileReader.GetNumberOfColumnsInProfile(jMarker);
+          vector<su2double> Inlet_Data_Interpolated (nCol_InletFile);
 
-          vector<su2double> Inlet_Values(nColumns);
+          /*--- Define Inlet Values vectors before and after interpolation (if needed) ---*/
+          vector<su2double> Inlet_Values(nCol_InletFile);
+          vector<su2double> Inlet_Interpolated(nColumns);
 
-          /*--- Loop through the nodes on this marker. ---*/
+          unsigned long nRows = profileReader.GetNumberOfRowsInProfile(jMarker);
 
-          for (iVertex = 0; iVertex < geometry[MESH_0]->nVertex[iMarker]; iVertex++) {
+          /*--- Pointer to call Set and Evaluate functions. ---*/
+          vector<C1DInterpolation*> interpolator (nColumns);
 
-            iPoint   = geometry[MESH_0]->vertex[iMarker][iVertex]->GetNode();
-            Coord    = geometry[MESH_0]->node[iPoint]->GetCoord();
-            min_dist = 1e16;
+          /*--- Object to call Corrected Inlet Values and Print Interpolated Data functions ---*/
+          C1DInterpolation *corrector = nullptr;
 
-            /*--- Find the distance to the closest point in our inlet profile data. ---*/
+          switch(config->GetKindInletInterpolationFunction()){
 
-            for (iRow = 0; iRow < profileReader.GetNumberOfRowsInProfile(jMarker); iRow++) {
+            case (NONE):
+            Interpolate = false;
+            break;
 
-              /*--- Get the coords for this data point. ---*/
+            case (ONED_AKIMASPLINE_SPANWISE):
```

Comment:
I tried to bring it as close as possible to what you suggested. Now my constructor is calling the SetSpline functions.